### DISCUSSION TITLE
CRYPTO_set_mem_functions always failing

### DISCUSSION THREAD

**Discussion by nagu1137** [openssl/openssl#25816] [Q&A] [answered]
CRYPTO_set_mem_functions is awlays returning failure 
we are using 3.1.3 openssl
any pointers to check?

**Comment by mattcaswell** [comment, marked as answer]
The only real reason that this function can fail is if `CRYPTO_malloc` has already been called (either directly by your application, or more likely because some other OpenSSL function has called it).  Once `CRYPTO_malloc` has been called once then you are no longer able to call `CRYPTO_set_mem_functions`. I recommend moving your call of this function as early as possible. It really should be the first OpenSSL function you call.
